client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import HeaderCake from 'components/header-cake';
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+import FilePicker from 'components/file-picker';
+import DropZone from 'components/drop-zone';
+import { localize } from 'i18n-calypso';
+import notices from 'notices';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:themes:theme-upload' );
+
+class Upload extends React.Component {
+
+	onFileSelect = ( files ) => {
+		const { translate } = this.props;
+		const errorMessage = translate( 'Please drop a single zip file' );
+
+		if ( files.length !== 1 ) {
+			notices.error( errorMessage );
+			return;
+		}
+		// Supplied param can be an array or a FileList
+		const file = files[ 0 ] || files.item( 0 );
+		if ( file.type !== 'application/zip' ) {
+			notices.error( errorMessage );
+			return;
+		}
+		debug( 'zip file:', file );
+	}
+
+	render() {
+		const { translate } = this.props;
+		const uploadPromptText = translate(
+			'Do you have a custom theme to upload to your site?'
+		);
+		const uploadInstructionsText = translate(
+			"Make sure it's a single zip file, and upload it here."
+		);
+		const dropText = translate(
+			'Drop files or click here to upload'
+		);
+
+		return (
+			<Main>
+				<HeaderCake onClick={ page.back }>{ translate( 'Upload theme' ) }</HeaderCake>
+				<Card>
+					<span className="theme-upload__prompt">{ uploadPromptText }</span>
+					<span className="theme-upload__instructions">{ uploadInstructionsText }</span>
+					<div className="theme-upload__dropzone">
+						<DropZone onFilesDrop={ this.onFileSelect } />
+						<FilePicker accept="application/zip" onPick={ this.onFileSelect } >
```

Comment:
> FilePicker is triggered by clicking anywhere inside the dashed area.

Hah. I was mistaking the dashed area for the DropZone.

Something's still a bit funny, tho -- Isn't the DropZone normally supposed to encompass the entire screen? Cause if I d'n'd a zip file onto my browser tab, it lights up only within the dashed area. If I drop the file outside, instead of being "uploaded" (no console log), it's being downloaded by the browser to my Downloads folder.